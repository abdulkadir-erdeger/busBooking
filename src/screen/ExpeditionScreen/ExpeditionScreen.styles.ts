import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: { flex: 1 },
  travelInfoCard: { backgroundColor: "#f9a825", padding: 15 },
  infoCitys: { flexDirection: "row", alignItems: "center" },
  cityText: { marginRight: 5, fontSize: 20, fontWeight: "bold" },
});
