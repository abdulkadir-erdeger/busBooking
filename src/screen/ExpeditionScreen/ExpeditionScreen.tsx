import { View, Text, FlatList } from "react-native";
import { useRoute } from "@react-navigation/native";
import { FontAwesome5 } from "@expo/vector-icons";
import styles from "./ExpeditionScreen.styles";
import TravelInfoCard from "../../component/TravelInfoCard";
import axios from "axios";
import { ProfileScreenRouteProp } from "../../types";
import React, { useEffect, useState } from "react";

const ExpeditionScreen = () => {
  const route = useRoute<ProfileScreenRouteProp>();
  const data: any = route?.params;
  const [travelData, setTravelData] = useState();

  useEffect(() => {
    axios
      .get("http://10.0.2.2:5000/travelData/")
      .then((res) => setTravelData(res.data))
      .catch((err) => console.log(err));
  }, []);

  return (
    <View style={styles.container}>
      <View style={styles.travelInfoCard}>
        <View style={styles.infoCitys}>
          <Text style={styles.cityText}>{data?.kalkÄ±s.toUpperCase()}</Text>
          <FontAwesome5
            style={styles.cityText}
            name="long-arrow-alt-right"
            size={24}
            color="black"
          />
          <Text style={styles.cityText}>{data?.varis.toUpperCase()}</Text>
        </View>
        <View style={styles.infoCitys}>
          <Text>{data?.haftaninGunu}, </Text>
          <Text>{data?.gun} </Text>
          <Text>{data?.ay} </Text>
          <Text style={{ textTransform: "capitalize" }}>{data?.yasDurum}</Text>
        </View>
      </View>
      <FlatList
        data={travelData}
        renderItem={({ item }) => <TravelInfoCard x={item} />}
        keyExtractor={(_, index) => index.toString()}
      />
    </View>
  );
};

export default ExpeditionScreen;
