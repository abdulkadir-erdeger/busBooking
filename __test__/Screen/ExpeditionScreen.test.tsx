import React from "react";
import { render, screen } from "@testing-library/react-native";
import ExpeditionScreen from "../../src/screen/ExpeditionScreen";

const mockedNavigate = jest.fn();
const mockedRoute = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useRoute: () => ({
      route: mockedRoute,
    }),
    useNavigation: () => ({
      navigation: mockedNavigate,
    }),
  };
});

test("render correctly", () => {
  render(<ExpeditionScreen />);
  expect(screen.toJSON()).toMatchSnapshot();
});
