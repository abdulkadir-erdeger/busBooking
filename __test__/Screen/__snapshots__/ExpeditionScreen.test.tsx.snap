// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f9a825",
        "padding": 15,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginRight": 5,
          }
        }
      />
      <Text />
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginRight": 5,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text>
        , 
      </Text>
      <Text>
         
      </Text>
      <Text>
         
      </Text>
      <Text
        style={
          {
            "textTransform": "capitalize",
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    data={
      [
        {
          "fiyat": "110,00",
          "id": 1,
          "inisSaati": "13:00",
          "inisYeri": "Ankara/Otogarı",
          "kalkisSaati": "7:15",
          "kalkisYeri": "Istanbul/Ümraniye Dudulu",
          "ozellikler": "",
          "seferBilgileri": "5:45",
        },
        {
          "fiyat": "110,00",
          "id": 1,
          "inisSaati": "15:00",
          "inisYeri": "Ankara/Otogarı",
          "kalkisSaati": "8:00",
          "kalkisYeri": "Istanbul/Esenler Otogarı",
          "ozellikler": "",
          "seferBilgileri": "7:00",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderRadius": 10,
              "borderWidth": 1,
              "flex": 1,
              "margin": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View>
              <Text
                testID="kalkisSaati"
              >
                7:15
              </Text>
              <Text
                testID="inisSaati"
              >
                13:00
              </Text>
            </View>
            <Text />
            <View>
              <Text
                testID="kalkisYeri"
              >
                Istanbul/Ümraniye Dudulu
              </Text>
              <Text
                testID="inisYeri"
              >
                Ankara/Otogarı
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "backgroundColor": "lightgrey",
                "height": 2,
                "marginVertical": 10,
              }
            }
          />
          <View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text />
                  <Text
                    style={
                      {
                        "marginLeft": 5,
                      }
                    }
                  >
                    Sefer Bilgileri | 
                    5:45
                     saat
                  </Text>
                </View>
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "paddingVertical": 5,
                    }
                  }
                >
                  <Text />
                  <Text />
                  <Text />
                  <View
                    style={
                      {
                        "backgroundColor": "lightgrey",
                        "marginHorizontal": 3,
                        "width": 1,
                      }
                    }
                  />
                  <Text />
                  <Text />
                  <Text />
                </View>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "#1C3879",
                    "borderRadius": 16,
                    "opacity": 1,
                    "padding": 10,
                    "paddingHorizontal": 30,
                  }
                }
                testID="buttonNavigate"
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                    }
                  }
                  testID="fiyat"
                >
                  ₺ 
                  110,00
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderRadius": 10,
              "borderWidth": 1,
              "flex": 1,
              "margin": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View>
              <Text
                testID="kalkisSaati"
              >
                8:00
              </Text>
              <Text
                testID="inisSaati"
              >
                15:00
              </Text>
            </View>
            <Text />
            <View>
              <Text
                testID="kalkisYeri"
              >
                Istanbul/Esenler Otogarı
              </Text>
              <Text
                testID="inisYeri"
              >
                Ankara/Otogarı
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "backgroundColor": "lightgrey",
                "height": 2,
                "marginVertical": 10,
              }
            }
          />
          <View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text />
                  <Text
                    style={
                      {
                        "marginLeft": 5,
                      }
                    }
                  >
                    Sefer Bilgileri | 
                    7:00
                     saat
                  </Text>
                </View>
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "paddingVertical": 5,
                    }
                  }
                >
                  <Text />
                  <Text />
                  <Text />
                  <View
                    style={
                      {
                        "backgroundColor": "lightgrey",
                        "marginHorizontal": 3,
                        "width": 1,
                      }
                    }
                  />
                  <Text />
                  <Text />
                  <Text />
                </View>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "#1C3879",
                    "borderRadius": 16,
                    "opacity": 1,
                    "padding": 10,
                    "paddingHorizontal": 30,
                  }
                }
                testID="buttonNavigate"
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                    }
                  }
                  testID="fiyat"
                >
                  ₺ 
                  110,00
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
